### PR TITLE
Cleare varnish cache when simple product goes out of stock

### PR DESCRIPTION
**Pull request by tmotyl**
Listen to the inventory change event and purge product and category
tags when product status has changed.

### CODE REVIEW

**Comment by tmotyl** [comment]
this is a part 2 of the splitted pull request https://github.com/AOEpeople/Aoe_Static/pull/7

**Comment by tmotyl** [comment]
ping

File: app/code/community/Aoe/Static/Model/Observer.php [hunk 1]
```diff
@@ -282,6 +282,36 @@ public function catalogCategorySaveCommitAfter($observer)
...
+     * Observer for cataloginventory_stock_item_save_after
+     * checking for product going out of stock
+     *
+     * This is needed for catching product availability change after it was sold out
+     *
+     * @param Varien_Event_Observer $observer
+     */
+    public function productOutOfStock($observer) {
+        /** @var $category Mage_CatalogInventory_Model_Stock_Item */
+        $stockItem = $observer->getItem();
```
Comment (LeeSaferite) [inline app/code/community/Aoe/Static/Model/Observer.php]: Add an 'instanceof' check here please.

File: app/code/community/Aoe/Static/Model/Observer.php [hunk 2]
```diff
@@ -282,6 +282,36 @@ public function catalogCategorySaveCommitAfter($observer)
...
+            && $stockItem->getIsInStock() != $originalStockData
+            && $stockItem->getProductId() > 0)
+        || $stockItem->getStockStatusChangedAuto()
+        ) {
+            $tagsToPurge = array();
+            /** @var $helper Aoe_Static_Helper_Data */
+            $helper = Mage::helper('aoestatic');
+            $tagsToPurge[] = 'product-' . $stockItem->getProductId();
+            $categories = Mage::getModel('catalog/product')->setId($stockItem->getProductId())->getCategoryIds();
+            foreach($categories as $category) {
```
Comment (LeeSaferite) [inline app/code/community/Aoe/Static/Model/Observer.php]: Super minor nit, but wouldn't `$categories` be better named `$categoryIds`?

**Pull request merged**